### ISSUE TITLE
Update libnetcdf pin

### ISSUE THREAD

**Issue by kmuehlbauer** [conda-forge/conda-forge-pinning-feedstock#353]
libnetcdf is currently pinned to 4.7.1. As 4.7.3 is already out, could the pin be updated? Or would it be possible to pin to just 4.7?

**Comment by ocefpaf** [comment]
> libnetcdf is currently pinned to 4.7.1. As 4.7.3 is already out, could the pin be updated? Or would it be possible to pin to just 4.7?

I believe we have a x.x pin but they broke ABI (changed the soname) from 4.6.2 to 4.6.3 and we decided to go with x.x.x for safety. While the 4.7 series seems OK, see https://abi-laboratory.pro/?view=timeline&l=netcdf, I'm afraid of pinning to x.x again b/c we never know what will happen in the next release. Also, upstream explicitly mentioned that do not follow or promise SEMVER.

TL;DR we should send a migration to 4.7.3.

**Comment by kmuehlbauer** [comment]
@ocefpaf what is the procedure to do so?

**Comment by isuruf** [comment]
> TL;DR we should send a migration to 4.7.3.

Or hot fix the upper bound.

**Comment by ocefpaf** [comment]
> Or hot fix the upper bound.

I would discourage hotfixes. They are really hard to follow from a reproducibility point of view.
